std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,42 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayEq[@sp A: Eq]: Eq[Array[A]] = new ArrayEq[A]
```

Comment:
The motivation was that byte boxing, which uses an array lookup table, was much faster than int boxing (which allocates), so it wasn't worth the extra methods.

But I guess `Eq` is still `@sp`, to keeping thes one `@sp` is fine. Actually, I guess `PartialOrder` and `Order` are `@sp` too, so, that's fine to stay consistent with them.

I think for Semigroup and it's subclasses we went with just the 4 and 8 byte types.